compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -763,8 +764,11 @@ fn dump_vtable_entries<'tcx>(
     trait_ref: ty::PolyTraitRef<'tcx>,
     entries: &[VtblEntry<'tcx>],
 ) {
-    let msg = format!("vtable entries for `{}`: {:#?}", trait_ref, entries);
-    tcx.sess.struct_span_err(sp, &msg).emit();
+    tcx.sess.emit_err(DumpVTableEntries {
+        span: sp,
+        trait_ref: trait_ref.to_string(),
```

Comment:
I'd appreciate it if you could try to implement `IntoDiagnosticArg` for some of the field types and use them directly (for all the errors). You might be able to add them [here](https://github.com/rust-lang/rust/blob/230a8ee364b693c286a2187e56239cd2cc3be318/compiler/rustc_errors/src/diagnostic.rs#L76-L93).

Reply:
I tried to add these types [there](https://github.com/rust-lang/rust/blob/230a8ee364b693c286a2187e56239cd2cc3be318/compiler/rustc_errors/src/diagnostic.rs#L76-L93) but I was not able to import the crates with the types from `compiler/rustc_errors/src/diagnostic.rs`

Reply:
If you can't add them in `rustc_errors` then add a `IntoDiagnosticImpl` where they are defined, if that's possible.